File: 06 - hashSet/java/src/ColecaoHash/Aluno.java
package ColecaoHash;

import java.util.Objects;

public class Aluno {
	int matricula;
	String nome;
	
	public int getMatricula() {
		return matricula;
	}

	public void setMatricula(int matricula) {
		this.matricula = matricula;
	}

	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public Aluno(int matricula, String nome) {
		super();
		this.matricula = matricula;
		this.nome = nome;
	}

	@Override
	public String toString() {
		return "Aluno [matricula=" + matricula + ", nome=" + nome + "]";
	}

	@Override
	public int hashCode() {
		return Objects.hash(matricula);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Aluno other = (Aluno) obj;
		return matricula == other.matricula;
	}
	
	
	
	

}

File: 05 - filas/cSharp/Aluno.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Fila
{
    internal class Aluno
    {
        public int Matricula { get; set; }
        public string Nome { get; set; }

        public Aluno(int matricula, string nome)
        {
            this.Matricula= matricula;
            this.Nome = nome;
        }
        
        public override string ToString()
        {
            return "Aluno [matricula=" + Matricula + ", nome=" + Nome + "]";
        }

        public override bool Equals(object obj)
        {
            Aluno other = (Aluno)obj;
            if (Matricula != other.Matricula)
                return false;
            if (!Nome.Equals(other.Nome))
                return false;
            return true;
        }
    }
}
